packages/flutter_test/lib/src/leak_tracking.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:leak_tracker/leak_tracker.dart';
+
+/// Handler to collect leak information.
+///
+/// Used by [LeakTrackingFlutterTestConfig.onLeaks].
+/// The parameter [leaks] contains details about found leaks.
+typedef LeaksCallback = void Function(Leaks leaks);
+
+const String _nonBackwardCompatibleApiWarning = 'Backward compatibility of this API is not guaranteed.';
+
+/// Configuration for leak trecking in unit tests.
+///
+/// The configuration is needed only for test debugging,
+/// not for regular test run.
+///
+/// Do not submit usage of the constructor to the repository,
+/// because backward compatibility of its API is not guaranteed.
+@Deprecated(_nonBackwardCompatibleApiWarning)
+class LeakTrackingFlutterTestConfig {
+  /// Creates a new instance of [LeakTrackingFlutterTestConfig].
+  @Deprecated(_nonBackwardCompatibleApiWarning)
```

Comment:
It feels odd to check in deprecated code intentionally. @goderbauer @dnfield is there anywhere else in the framework where we do this / is this an acceptable pattern? I know in other places we have `_kDebug*` flags that can be flipped manually or via a service extension.

I would worry that users would ignore the deprecation warnings and check in usages of `LeakTrackingFlutterTestConfig` to their repos anyway.

Reply:
This is not a pattern we have elsewhere and we shouldn't establish it. I am afraid, this is just teaching people to ignore deprecation warnings, which is not in our interest. It is also not in line with our breaking change policy: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

Reply:
General practice would be to evolve the API as a package and then pull it into the framework when it is reasonably stable.